lib/Site.js

```diff
@@ -185,6 +188,39 @@ Site.prototype.collectBaseUrl = function () {
   return Promise.resolve();
 };
 
+/**
+ * Collects the user defined variables map in the site/subsites
+ */
+Site.prototype.collectUserDefinedVariablesMap = function () {
+  // The key is the base directory of the site/subsites,
+  // while the value is user defined variable map for that particular subsite
+  this.userDefinedVariablesMap = {};
+
+  Object.keys(this.baseUrlMap).forEach((base) => {
+    const userDefinedVariables = {};
+    const definedVariablePath = path.resolve(base, USER_VARIABLES_PATH);
+    let content;
+    try {
+      content = fs.readFileSync(definedVariablePath, 'utf8');
+    } catch (e) {
+      content = '';
+      logger.info(e.message);
+    }
+    const $ = cheerio.load(content);
+    $.root().children().each(function () {
+      const id = $(this).attr('id');
+      const html = $(this).html();
+      userDefinedVariables[id] = html;
+    });
+
+    // At this point, baseUrl has not been calculated.
```

Comment:
Actually, if you're iterating `this.baseUrlMap`, then `baseUrl = this.baseUrlMap[base]`.
But I'm not sure if assigning `baseUrl` here is a good idea, so let's just remove this line.